coopertunes/models/MelGan.py

```diff
@@ -0,0 +1,150 @@
+import torch.nn as nn
+import torch.nn.functional as F
+import torch
+from librosa.filters import mel as librosa_mel_fn
+from torch.nn.utils import weight_norm
+import numpy as np
+
+from coopertunes.hparams import HParams
+
+def weights_init(m):
+    classname = m.__class__.__name__
+    if classname.find("Conv") != -1:
+        m.weight.data.normal_(0.0, 0.02)
+    elif classname.find("BatchNorm2d") != -1:
+        m.weight.data.normal_(1.0, 0.02)
+        m.bias.data.fill_(0)
+
+def WNConv1d(*args, **kwargs):
+    return weight_norm(nn.Conv1d(*args, **kwargs))
+
+def WNConvTranspose1d(*args, **kwargs):
+    return weight_norm(nn.ConvTranspose1d(*args, **kwargs))
+
+
+class ResnetBlock(nn.Module):
+    def __init__(self, dim, dilation=1):
+        super().__init__()
+        self.block = nn.Sequential(
+            nn.LeakyReLU(0.2),
+            nn.ReflectionPad1d(dilation),
+            WNConv1d(dim, dim, kernel_size=3, dilation=dilation),
+            nn.LeakyReLU(0.2),
+            WNConv1d(dim, dim, kernel_size=1),
+        )
+        self.shortcut = WNConv1d(dim, dim, kernel_size=1)
+
+    def forward(self, x):
+        return self.shortcut(x) + self.block(x)
+
+
+class MelGanGenerator(nn.Module):
```

Comment:
Please add inference if possible. I mean for usage after training there is no need for returning feature map

Reply:
I see in MelSpec Vae inference is just the decoding part of architecture. I believe in case of MelGAN eval inference and training inference looks the same